internal/protocoltest/awsrestjson/endpoints.go

```diff
@@ -45,6 +45,26 @@ func resolveDefaultEndpointConfiguration(o *Options) {
 	o.EndpointResolver = NewDefaultEndpointResolver()
 }
 
+// EndpointResolverFromURL returns an EndpointResolver configured using the
+// provided endpoint url. By default, the resolved endpoint resolver uses the
+// client region as signing region. You can provide functional options to configure
+// endpoint values for the resolved endpoint.
+func EndpointResolverFromURL(url string, optFns ...func(*aws.Endpoint)) EndpointResolver {
+	e := aws.Endpoint{URL: url}
+	for _, fn := range optFns {
+		fn(&e)
+	}
+
+	return EndpointResolverFunc(
+		func(region string, options EndpointResolverOptions) (aws.Endpoint, error) {
+			if len(e.SigningRegion) == 0 {
+				e.SigningRegion = region
+			}
+			return e, nil
+		},
+	)
+}
+
```

Comment:
Generation of these should be updated for the Endpoint's `Source` member being set to `Custom`